### ISSUE TITLE
Typescript <> style type assertions cause JSX parsing errors.

### ISSUE THREAD

**Issue by Benalink** [vuejs/eslint-plugin-vue#919]
**Tell us about your environment**	
* **ESLint version:** 5.16.0	
* **eslint-plugin-vue version:** 3.8.0	
* **Node version:**	10.15.3

**Please show your full configuration:**

```json
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/essential",
      "@vue/prettier",
      "@vue/typescript"
    ],
    "rules": {},
    "parserOptions": {
      "parser": "@typescript-eslint/parser"
    }
  }
```

**What did you do?**	

```vue
<script lang="ts">
let repro = function() {
    let anyWindow = <any>window;
}
</script>
```

**What did you expect to happen?**
No parsing errors are thrown

**What actually happened?**

```
> vue-cli-service lint "src/components/EslintFalsePositive.vue"

error: Parsing error: JSX element 'any' has no corresponding closing tag at src\components\EslintFalsePositive.vue:3:21:
  1 | <script lang="ts">
  2 | let repro = function() {
> 3 |     let anyWindow = <any>window
    |                     ^
  4 | }
  5 | </script>
  6 |


1 error found.
```
If I missed anything or you want the complete project I have included a new cli project with the only addition being the offending file https://github.com/Benalink/eslint-parser-bug

**Comment by kargpfen** [comment]
I have the same problem, have you any solution?

Greets

**Comment by Benalink** [comment]
Hi @kargpfen 

It's not a proper solution but as a workaround I've been using `let anyWindow = window as any` style type assertions instead of the angle bracket style.

**Comment by Tomassito** [comment]
> Hi @kargpfen
> 
> It's not a proper solution but as a workaround I've been using `let anyWindow = window as any` style type assertions instead of the angle bracket style.

Actually this is the official 'workaround' ;) 
https://www.typescriptlang.org/docs/handbook/jsx.html#the-as-operator

**Comment by armano2** [comment]
if you intend of using `<>`  casting you should consider turning off JSX
```
ecmaFeatures: {
  jsx: false
},
```
https://eslint.org/docs/user-guide/configuring#specifying-parser-options

and make sure to have it turned off in tsconfig
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/README.md#parseroptionsecmafeaturesjsx
